src/Ansatz.jl

```diff
@@ -0,0 +1,485 @@
+"""
+@kwmethod function inds(tn::AbstractAnsatz; bond)
+    (site1, site2) = bond
+    @assert site1 ∈ sites(tn) "Site $site1 not found"
+    @assert site2 ∈ sites(tn) "Site $site2 not found"
+    @assert site1 != site2 "Sites must be different"
+    @assert has_edge(tn, site1, site2) "Sites must be neighbors"
+
+    tensor1 = tensors(tn; at=site1)
+    tensor2 = tensors(tn; at=site2)
+
+    isdisjoint(inds(tensor1), inds(tensor2)) && return nothing
+    return only(inds(tensor1) ∩ inds(tensor2))
+end
+
+"""
+    tensors(tn::AbstractAnsatz; bond)
+
+Return the [`Tensor`](@ref) in a virtual bond between two [`Site`](@ref)s in a [`AbstractAnsatz`](@ref) Tensor Network.
+
+# Notes
+
+  - If the [`AbstractAnsatz`](@ref) Tensor Network is in the canonical form, Tenet stores the Schmidt coefficients of the bond in a vector connected to the bond hyperedge between the two sites and the vector.
+  - If the bond contains no Schmidt coefficients, this method will throw a `MissingSchmidtCoefficientsException`.
+"""
+@kwmethod function tensors(tn::AbstractAnsatz; bond)
+    vind = inds(tn; bond)
+    tensor = tensors(tn, [vind]) do vinds, indices
+        indices == vinds
+    end
+    isempty(tensor) && throw(MissingSchmidtCoefficientsException(bond))
+    return only(tensor)
+end
+
+@kwmethod function tensors(tn::AbstractAnsatz; between)
+    Base.depwarn(
+        "`tensors(tn; between)` is deprecated, use `tensors(tn; bond)` instead.",
+        ((Base.Core).Typeof(tensors)).name.mt.name,
+    )
+    return tensors(tn; bond=between)
+end
+
+"""
+    contract!(tn::AbstractAnsatz; bond)
+
+Contract the virtual bond between two [`Site`](@ref)s in a [`AbstractAnsatz`](@ref) Tensor Network.
+"""
+@kwmethod contract!(tn::AbstractAnsatz; bond) = contract!(tn, inds(tn; bond))
+
+canonize(tn::AbstractAnsatz, args...; kwargs...) = canonize!(deepcopy(tn), args...; kwargs...)
+canonize_site(tn::AbstractAnsatz, args...; kwargs...) = canonize_site!(deepcopy(tn), args...; kwargs...)
+
+"""
+    truncate(tn::AbstractAnsatz, bond; threshold = nothing, maxdim = nothing)
+
+Like [`truncate!`](@ref), but returns a new tensor network instead of modifying the original one.
+"""
+truncate(tn::AbstractAnsatz, args...; kwargs...) = truncate!(deepcopy(tn), args...; kwargs...)
+
+"""
+    truncate!(tn::AbstractAnsatz, bond; threshold = nothing, maxdim = nothing)
+
+Truncate the dimension of the virtual `bond`` of an [`Ansatz`](@ref) Tensor Network by keeping only the `maxdim` largest Schmidt coefficients or those larger than`threshold`.
+
+# Notes
+
+  - Either `threshold` or `maxdim` must be provided. If both are provided, `maxdim` is used.
+  - The bond must contain the Schmidt coefficients, i.e. a site canonization must be performed before calling `truncate!`.
+"""
+function truncate!(tn::AbstractAnsatz, bond; threshold=nothing, maxdim=nothing)
+    @assert isnothing(maxdim) ⊻ isnothing(threshold) "Either `threshold` or `maxdim` must be provided"
+
```

Comment:
```suggestion
    @assert form(ψ) != NonCanonical() "A `NonCanonical` tensor network can not be truncated"

```

Reply:
Again, I don't agree with this affirmation. You CAN truncate even if it's far from optimal, and in many cases it works.